rust/operator-binary/src/controller.rs

```diff
@@ -1161,3 +1022,109 @@ fn liveness_probe(trino: &TrinoCluster) -> Probe {
         ..Probe::default()
     }
 }
+
+fn create_tls_volume(volume_name: &str, tls_secret_class: Option<&TlsSecretClass>) -> Volume {
+    let secret_class_name = tls_secret_class
+        .map(|t| t.secret_class.as_str())
+        .unwrap_or(TLS_DEFAULT_SECRET_CLASS);
+    VolumeBuilder::new(volume_name)
+        .ephemeral(
+            SecretOperatorVolumeSourceBuilder::new(secret_class_name)
+                .with_pod_scope()
+                .with_node_scope()
+                .build(),
+        )
+        .build()
+}
+
+fn tls_volume_mounts(
+    trino: &TrinoCluster,
+    pod_builder: &mut PodBuilder,
+    cb_prepare: &mut ContainerBuilder,
+    cb_trino: &mut ContainerBuilder,
+    s3_connection: Option<&S3ConnectionSpec>,
+) -> Result<()> {
```

Comment:
To be honest, this whole function feels quite incoherent, closer to "this is what we happened to need for this one use case" than something I'd feel comfortable documenting and reusing.

Reply:
Yeah thats for reducing the code in the build_stateful_set method